Mod2/Lessons/Week5/DatabaseMigrations.md

```diff
@@ -0,0 +1,135 @@
+# Database Migrations
+
+We've learned the foundations of our ORM, Entity Framework. Today we're going to dive deeper into Entity Framework and what it would be like to work with a database on the job.
+
+## Learning Goals
+- Explore business decisions that drive database changes
+- Practice making database changes to add and remove columns
+- Learn how to create seed data
+
+## Warmup
+
+Let's do a chat waterfall to hear the questions everyone had after reading the migrations documentation:
+1. What's one question you currently have about migrations?
+
+> With your partner, discuss your answers to the following questions:
+> 1. How would you describe the process of evolving your model to add a new column?
+> 1. What's one thing you learned about migrations from reading this documentation?
+
+### Business Decisions that Drive DB Changes
+
+Adding a column is one type of database change you may be asked to implement on the job. There are many other types of DB changes you may run into such as modifying a column, deleting a column, or adding a table.
+
+Here is a real-world example from Zoe: My first task after joining the real estate company Orchard was to allow our real estate agents to record if their meeting would take place virtually or in person. Our agents used to have all meetings in person, but because of COVID they started offering virtual appointments and we wanted to track that data in our database. In this situation, I needed to add a new column to the database.
+
+> With you group: Brainstorm business decisions that would cause a developer to modify the database table. What exactly would need to change in the database?
+
+Here are some sentence starters to help you get started!
+
+**One business decision that probably to a database change was when __________ decided to __________. The developers probably needed to update the database by ______________________.**
+
+### Implementing DB Changes
+
+Let's work together to make a DB Change in our plant app. We now want to keep track of when each plant was last watered so that we can remind our users when to water their plants. Let's call the column `last_watered` and make it type DateTime.
```

Comment:
Love that we are building on former code!  Maybe at the beginning of the lesson we need an instructor note to have students open this app?

Reply:
We might need some more instructions on how to create this migration in the lesson.  I'm thinking about when a student misses class and needs to catch up... or an instructor is sick and someone needs to step in...

I like the walkthrough feel though!

Reply:
Good call, I'll add an instuctor note to clarify what I'm thinking. The walkthrough it all in this link. https://learn.microsoft.com/en-us/ef/core/managing-schemas/migrations/?tabs=dotnet-core-cli#evolving-your-model

I'm imagining the instructor would open that page next to visual studio and walk through using that as a reference for the commands.